qiskit/circuit/library/generalized_gates/multicontrol_arb_gate.py

```diff
@@ -0,0 +1,265 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""multicontrol single qubit unitary gate."""
+
+from typing import Union, List, Optional
+import numpy
+from scipy.linalg import fractional_matrix_power
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.circuit.controlledgate import ControlledGate
+from qiskit.exceptions import QiskitError
+from qiskit.circuit._utils import _ctrl_state_to_int, _compute_control_matrix
+from qiskit.extensions.quantum_initializer.squ import SingleQubitUnitary
+
+
+class MCU2Gate(ControlledGate):
```

Comment:
My suggestion is to create a different PR with the MCU2Gate because this class is for "any multicontrol version of any single qubit unitary matrix (gate)" and this PR is about multicontrolled rotation gates.